r-package/policytree/R/hybrid_policy_tree.R

```diff
@@ -0,0 +1,201 @@
+#' Hybrid tree search (experimental)
+#'
+#' Finds a depth k tree by looking ahead l steps.
+#'
+#'
+#' Builds deeper trees by iteratively using exact tree search to look ahead l splits. For example,
+#' with `depth = 3` and `search.depth = 2`, the first split is determined by a depth 2 exact tree,
+#' and two new depth 2 trees are fit in the immediate children, leading to a total depth of 3.
+#' Note that the resulting tree may be shallower than the specified `depth` depending on
+#' whether leaf nodes were pruned or not.
+#' This algorithm scales with some coefficient multiple of the runtime of a `search.depth` `policy_tree`.
+#'
+#' The algorithm: desired depth is given by `depth`. Each node is split using exact tree search
+#' with depth  = `search.depth`. When we reach a node where the current level + `search.depth` is equal to `depth`,
```

Comment:
and the root split is kept from this exact tree search. The immediate children are further split using exact tree search.

Reply:
This is OK?

```
For example, with depth = 3 and search.depth = 2, the root split is determined by a depth 2 exact tree, and two new depth 2 trees are fit in the immediate children using exact tree search, leading to a total depth of 3.
```

Reply:
Yes.